pytorch_to_returnn/torch/nn/modules/rnn.py

```diff
@@ -11,12 +12,42 @@
 from ..parameter import Parameter
 from ...tensor import Tensor
 from .. import init
+from ....naming import Naming, TensorEntry
 
 
-class PackedSequence:  # dummy -- not yet implemented...
+class BatchSizesDummy:
   pass
 
 
+Naming.IgnoredTypes.append(BatchSizesDummy)
+
+
+PackedSequence_ = namedtuple(
+  'PackedSequence',
+  ['data', 'batch_sizes', 'sorted_indices', 'unsorted_indices'],
+  defaults=[None] * 4)
+
+
+class PackedSequence(PackedSequence_):
+  """
+  Wrapper for torch's PackedSequence
+
+  We connect each PackedSequence with a batched RETURNN tensor. This currently contains some assumptions which might not
+  always be satisfied.
+  """
+
+  def get_batched_tensor(self) -> Tensor:
+    from .shape import FlattenBatch
+    naming = Naming.get_instance()
+    tensor_entry = naming.tensors[self.data]
+    assert isinstance(tensor_entry, TensorEntry)
+    for call_entry in tensor_entry.output_from_calls:
+      if isinstance(call_entry.module.module, FlattenBatch):
+        return call_entry.orig_inputs_args[0]
+    from .shape import UnflattenBatch
+    return UnflattenBatch()(self.data)
```

Comment:
Maybe we want to throw an exception and not do the `UnflattenBatch` here? Not sure...

Reply:
That's what you initially proposed, but I added this as a fallback. In the example that I'd like to convert, there is something like 

`h = pack_sequence(...); h_data = some_op(h.data); h = PackedSequence(h_data, h.batch_sizes)`

Then, `self.data` is not a direct output of `FlattenBatch`, so it cannot be found in the way above.